doc/source/whatsnew/v0.24.0.txt

```diff
@@ -196,6 +196,158 @@ Other Enhancements
 Backwards incompatible API changes
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+.. _whatsnew_0240.api_breaking.csv_line_terminator:
+
+`os.linesep` is used for ``line_terminator`` of ``DataFrame.to_csv``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+:func:`DataFrame.to_csv` now uses :func:`os.linesep` rather than ``'\n'``
+ for the default line terminator(:issue:`20353`).
+- This change only affects when running on Windows, where ``'\r\n'`` was used for line terminator
+even when ``'\n'`` was passed in ``line_terminator``.
+  - Strictly speeaking, all ``'\n'``s appear in data and line terminator of CSV were converted into ``'\r\n'``s.
+  - This problem was resolved by passing file object with ``newline='\n'`` option as output, rather than file name.
+
+Previous Behavior on Windows:
+
+.. code-block:: ipython
+
+  In [1]: import pandas as pd
+
+  In [2]: data = pd.DataFrame({
+     ...: "string_with_lf":["abc","d\nef","g\nh\n\ni"],
+     ...: "string_with_crlf":["abc","d\r\nef","g\r\nh\r\n\r\ni"]
+     ...: })
+
+  In [3]: data.to_csv("test.csv",index=False,line_terminator='\n')
+
+  In [4]: print(pd.read_csv("test.csv"))
+      string_with_lf       string_with_crlf
+  0              abc                    abc
+  1          d\r\nef              d\r\r\nef
+  2  g\r\nh\r\n\r\ni  g\r\r\nh\r\r\n\r\r\ni
+
+  In [5]: with open("test.csv", mode='rb') as f:
+     ...:     print(f.read())
+  b'string_with_lf,string_with_crlf\r\nabc,abc\r\n"d\r\nef","d\r\r\nef"\r\n"g\r\nh
```

Comment:
can make this simpler, just show [3] , [4]

Reply:
yes, the point is we want a small example that shows the most common case where a user would want to know what is changing. Too long and you lose the reader, too short and you don't know if the change applies to youl.